sentry-ruby/lib/sentry/transaction.rb

```diff
@@ -49,20 +45,9 @@ def to_hash
       hash
     end
 
-    def start_child(**options)
-      child_span = super
-      child_span.span_recorder = @span_recorder
-
-      if @sampled
-        @span_recorder.add(child_span)
-      end
-
-      child_span
-    end
-
     def deep_dup
       copy = super
-      copy.set_span_recorder
+      copy.set_span_recorder(@span_recorder.max_length)
```

Comment:
Hmmm this is interesting. What are the use cases for deep cloning a Span / Transaction, why do we implement this?

Reply:
I forgot the original motivation for it. but generally I want to avoid a mutation on copy being shared with its origin, like:

```ruby
      span = Sentry::Transaction.new(sampled: true)
      subject.set_span(span)
      copy = subject.dup

      span.start_child

      # the copy probably shouldn't sees the same recorder change, no?
      expect(copy.span.span_recorder.spans.count).to eq(1)
```

I've written several test cases to prevent sharing mutations between transaction/spans and their copies. so if we decide this is unnecessary and want to drop it, I prefer to do it in another PR.

Reply:
Definitely something for another PR, sorry for the distraction. It caught my attention since it is the first time I'm looking in more detail in this SDK.

> the copy probably shouldn't sees the same recorder change, no?

I don't know what I would expect... both possibilities could be okay depending on what it means to duplicate a span.

I understand and noticed how it is used in tests. In production code, nobody should probably be calling `Transaction.dup` or `Span.dup`?! ~~I am genuinely curious for the use cases of duplicating spans~~ I found one.

I see this is used when cloning a `Scope`. Unfortunately I think the behavior here is not clearly specified in the SDK specs. Python and JS do not perform a deep copy, only shallow, the recorder is shared (whether that is useful or not).